lion/io/database_parameters.h

```diff
@@ -52,37 +52,49 @@ inline void read_parameters(Xml_document& doc, const std::string& path, const st
 
     for (size_t i = 0; i < p.size(); ++i)
     {
+        Xml_element element = doc.get_element(path + p[i].name);
+
         switch (p[i].type)
         {
          case(Database_parameter_mutable::DOUBLE): 
-            *static_cast<scalar*>(p[i].address) = doc.get_element(path + p[i].name).get_value(scalar());
+            *static_cast<scalar*>(p[i].address) = element.get_value(scalar());
             break;
 
          case(Database_parameter_mutable::INT): 
-            *static_cast<int*>(p[i].address) = doc.get_element(path + p[i].name).get_value(int());
+            *static_cast<int*>(p[i].address) = element.get_value(int());
             break;
 
          case(Database_parameter_mutable::STD_VECTOR_DOUBLE): 
-            *static_cast<std::vector<scalar>*>(p[i].address) = doc.get_element(path + p[i].name).get_value(std::vector<scalar>());
+            *static_cast<std::vector<scalar>*>(p[i].address) = element.get_value(std::vector<scalar>());
             break;
 
          case(Database_parameter_mutable::VECTOR3): 
-            *static_cast<sVector3d*>(p[i].address) = doc.get_element(path + p[i].name).get_value(sVector3d());
+            *static_cast<sVector3d*>(p[i].address) = element.get_value(sVector3d());
             break;
 
          case(Database_parameter_mutable::MATRIX3X3): 
-            *static_cast<sMatrix3x3*>(p[i].address) = doc.get_element(path + p[i].name).get_value(sMatrix3x3());
+            *static_cast<sMatrix3x3*>(p[i].address) = element.get_value(sMatrix3x3());
             break;
 
          case(Database_parameter_mutable::AD): 
-            *static_cast<CppAD::AD<scalar>*>(p[i].address) = doc.get_element(path + p[i].name).get_value(scalar());
+            *static_cast<CppAD::AD<scalar>*>(p[i].address) = element.get_value(scalar());
             break;
 
          default:
             throw lion_exception("[ERROR] read_parameters -> parameter type was not recognized");
             break;
         }
         used_parameters[i] = true;
+        element.add_attribute<bool>("__unused__", false);
+
+        while(true)
```

Comment:
Not a fan of `while(true)`. Maybe use something like `while(element.has_parent())`